src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,52 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
 
+public class WorkWithFile {
+    
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    
     public void getStatistic(String fromFileName, String toFileName) {
-
+        String line;
+        StringBuilder stringBuilder = new StringBuilder();
+        int summaSuply = 0;
+        int summaBuy = 0;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName))) {
+            while ((line = bufferedReader.readLine()) != null) {
+                String[] string = line.split(",");
+                stringBuilder.append(string[0]).append(" ").append(string[1]).append("\n");
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        String[] newarray = stringBuilder.toString().split("\n");
+        stringBuilder = new StringBuilder();
+        
+        for (String s : newarray) {
+            String[] drop = s.split(" ");
+            if ((SUPPLY.equals(drop[0]))) {
+                summaSuply = summaSuply + Integer.parseInt(drop[1]);
+            } else {
+                summaBuy = summaBuy + Integer.parseInt(drop[1]);
+            }
+            
+        }
+        rewriterule(stringBuilder.append(SUPPLY).append(",").append(summaSuply)
+                .append("\n").append(BUY).append(",").append(summaBuy).append("\n")
+                .append("result").append(",").append(summaSuply - summaBuy).toString(), toFileName);
```

Comment:
Large methods is very difficult to read and understand. So methods should be short, between 5-15 lines. And it’s important for you to remember that one method should be responsible for one task.
https://mate-academy.github.io/jv-program-common-mistakes/java-core/builder-file/work-with-file#divide-program-logic